specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2022-10-01/machineLearningServices.json

```diff
@@ -1270,6 +1273,56 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/computes/{computeName}/updateSchedules": {
```

Comment:
>updateSchedules [](http://example.com/codeflow?start=166&length=15)

The ARM CRUD model would make this a PATCH request on the ComputeResource. POST can't be applied in templates and can't be managed via policy, so PATCH is much preferred.

Reply:
Yes, this is a PATCH if we think from compute perspective. However, we plan to make schedules an independent entity going forward. 

So, what this API is actually doing is "Creating" whatever schedule entities that are passed to it and attaching it to compute resource. The URI should actually not contain the verb (which confuses things) but this is how it has existed since a couple of years.

Reply:
I'm not following how that recommends POST over PATCH. This PR is creating a new action on a compute resource. I'm suggesting implementing it as a PATCH method instead of POST action for a few reasons. It belongs as part of the CRUD model for compute resources, which should not use POST. Ref: https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc010-use-put-replace-or-patch-partial-update-to-update-a-resource

If you are planning to implement separate CRUD operations for schedules in the future, the ARM guidance for that will be that the schedule resources should not be replicated in the parent resource at all. Ref: https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc019-rbac-violation-information-leak-when-resources-full-content-of-other-types-are-included-in-the-response